### PR TITLE
Removed the update CI target packages step.

### PR DESCRIPTION
**Pull request by sima-zhu**
This is the pre-requisite step of the target packages updates workflow.

The proposal:
1. Remove the update target packages step from release pipeline which avoid the race condition on updating one common file. (The PR is related to this step) We have to make this happen first, otherwise both release pipeline and nightly run pipeline will touch the package list file in docs repo.
2. Implement lang specific scripts and onboard to separate daily pipeline currently `docindex` to update the target packages nightly based on the release csv file in azure-sdk. 
3. Extract common parts out  from different languages, and move to eng/common

Test pipeline: 
JS: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=816606&view=logs&j=ceb84d66-6af3-5d95-9875-cd812f26b59e&t=22ab889e-d7a0-5021-aa42-c86bca1ff413
Check in code: https://github.com/MicrosoftDocs/azure-docs-sdk-node/commit/01eac4a1d9e590cf2bb59cb2b7fef4eb2b04edb0

JAVA: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=816602&view=results
Check in code:

### CODE REVIEW

**Review by danieljurek — APPROVED** [review]

File: eng/common/scripts/update-docs-ci.ps1 [hunk 1]
```diff
@@ -1,70 +0,0 @@
-#Requires -Version 6.0
```
Comment (weshaggard) [inline eng/common/scripts/update-docs-ci.ps1:1]: Isn't this script still useful but in our other automation process or will we need to rewrite it completely?

Comment (sima-zhu) [reply eng/common/scripts/update-docs-ci.ps1:1]: I did not realize we have any other automation depends on this. We can leave it here, but remove the step only.

Comment (weshaggard) [reply eng/common/scripts/update-docs-ci.ps1:1]: I'm not sure if we are using it elsewhere that was a question for you.

Comment (sima-zhu) [reply eng/common/scripts/update-docs-ci.ps1:1]: For my knowledge, it is only used for update target CI which is docs repo update. So once we want to remove the step, the script is not in use anymore.

**Review by weshaggard — APPROVED** [review]

File: eng/common/scripts/update-docs-ci.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/common/scripts/update-docs-ci.ps1:1]: Sounds good. I just didn't want to break anything and I also didn't want to remove it if you planned to add it back for new implementation we plan to run in the docindex pipelines.

**Pull request merged**